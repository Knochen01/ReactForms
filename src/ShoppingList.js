import React, { useState }  from 'react'
import Item from './Item'
import NewItemForm from './NewItemForm'
import { v4 as uuidv4 } from 'uuid'

const ShoppingList = () => {

    const INITIAL_STATE = [
        {id: uuidv4(),  name:"Peanut-Butter", qty: 2},
        {id: uuidv4(),  name:"Almond-Milk", qty: 1}
    ]

    const [items, setItems] = useState(INITIAL_STATE)
    const addItem = (newItem) => {
        setItems(items => [...items, {...newItem, id:uuidv4()}])
    }
    return (
        <div>
            <h3>ShoppingList</h3>
            < NewItemForm addItem={addItem} />
            {items.map(({id, name, qty}, idx) => <Item  id={id} name={name} qty={qty} key={id} /> )}
        </div>
    )
}
export default ShoppingList