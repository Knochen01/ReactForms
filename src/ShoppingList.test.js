import React from "react";
import { render ,fireEvent} from '@testing-library/react'
import ShoppingList from "./ShoppingList";

 
// SmokeTest 
it("should render on the screen", () => {
    render(<ShoppingList />)
})

// SnapshotTest
it("should match Snapshot", () => {
    const { asFragment } = render(<ShoppingList />)
    expect(asFragment()).toMatchSnapshot()
})

// Add new Item
it('should add new Item', () => {
   const { queryByText, getByLabelText } = render(<ShoppingList />)
   const btn = queryByText('Add Item')
   const input = getByLabelText("Product:")
   expect(queryByText("ProductName:Chocolate Milk")).not.toBeInTheDocument()
   fireEvent.change(input, { target: {value: "Chocolate Milk"} });
   fireEvent.click(btn)
   expect(queryByText("ProductName:Chocolate Milk")).toBeInTheDocument()

}) 