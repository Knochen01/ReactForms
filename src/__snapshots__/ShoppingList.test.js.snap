// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      ShoppingList
    </h3>
    <form>
      <label
        for="name"
      >
        Product:
      </label>
      <input
        id="name"
        name="name"
        placeholder="ItemName"
        type="text"
        value=""
      />
      <label
        for="qty"
      >
        Quantity: 
      </label>
      <input
        max="10"
        min="1"
        name="qty"
        type="range"
        value=""
      />
      <button>
        Add Item
      </button>
    </form>
    <ul>
      <li>
        ProductName:Peanut-Butter
      </li>
      <li>
        Quantity:2
      </li>
      <li />
    </ul>
    <ul>
      <li>
        ProductName:Almond-Milk
      </li>
      <li>
        Quantity:1
      </li>
      <li />
    </ul>
  </div>
</DocumentFragment>
`;
